src/lib/installer.ts

```diff
@@ -0,0 +1,225 @@
+		throw new Error('There were no attached assets to the choosen release. You will have to manually download the binaries.')
+
+	const asset = release.assets.find((asset: any) => new RegExp(`^protoc-gen-(?:openapi|swagger).+${platform}-x86_64(?:\.exe$|$)`).test(asset.name))
+
+	if (!asset)
+		throw new Error(`There is no compatible download with release tag '${tag}' for your os.\nThis should not have happened, contact the software publisher.`)
+
+	const bin = Buffer.from((await octokit.request(asset.browser_download_url)).data)
+
+	return new Promise<{ bin: Buffer, tag: string, exe: boolean }>((resolve, reject) => {
+		fs.access(path.join(__dirname, '../../bin'), (err) => {
+			const write = () => {
+				const res = () => resolve({ bin, tag: release.tag_name, exe: platform === 'windows' })
+
+				if (!tag)
+					// TODO: Write a path to the latest bin file instead of writing it twice.
+					fs.writeFile(path.join(__dirname, '../..', 'bin/bin_current' + (platform === 'windows' ? '.exe' : '')), bin, (err) => {
+						if (err) reject(err)
+						res()
+					})
+				else
+					res()
+			}
+
+			if (err)
+				fs.mkdir(path.join(__dirname, '../../bin'), (err) => {
+					if (err) reject(err)
+					write()
+				})
+			else
+				write()
+		})
+	})
+}
+
+/**
+ * Installs the specified binary. Returns path to the binary.
+ * 
+ * Can be used in combination with `downloadAsync()`
+ * 
+ * @param bin Buffer with the binary
+ * @param tag Release tag
+ * @param exe Whether the binary is .exe
+ * 
+ * @example
+ * ```typescript
+ * install(await downloadAsync(tag))
+ * ```
+ */
+export function install(bin: Buffer, tag: string, exe: string): string
+export function install(bin: { bin: Buffer, tag: string, exe: boolean }): string
+export function install(...args: any[]): string {
+	const [_bin, _tag, _exe] = args as [Buffer | { bin: Buffer, tag: string, exe: boolean }, string | undefined, boolean | undefined]
+
+	const tag = ('bin' in _bin ? _bin.tag : _tag) ?? ''
+	const bin = ('bin' in _bin ? _bin.bin : _bin) ?? ''
+	const exe = ('bin' in _bin ? _bin.exe : _exe) ?? false
+
+	const filename = 'bin/bin_' + tag + (exe ? '.exe' : '')
+
+	if (!fs.existsSync(path.join(__dirname, '../../bin')))
+		fs.mkdirSync(path.join(__dirname, '../../bin'))
+
+	fs.writeFileSync(path.join(__dirname, '../..', filename), bin)
+
+	return path.join(__dirname, '../..', filename)
+}
+
+/**
+ * Installs the specified binary asynchronously. Returns path to the binary.
+ * 
+ * Can be used in combination with `downloadAsync()`
+ * 
+ * @param bin Buffer with the binary
+ * @param tag Release tag
+ * @param exe Whether the binary is .exe
+ * 
+ * @example
+ * ```typescript
+ * await installAsync(await downloadAsync(tag))
+ * ```
+ */
+export function installAsync(bin: Buffer, tag: string, exe: string): Promise<string>
+export function installAsync(bin: { bin: Buffer, tag: string, exe: boolean }): Promise<string>
+export function installAsync(...args: any[]) {
```

Comment:
Why don't we just provide _one_ install method (the async version preferably)?

Reply:
For synchronous function. Often used by CLI's.